gui/src/components/Taipy/PaginatedTable.spec.tsx

```diff
@@ -269,4 +296,209 @@ describe("PaginatedTable Component", () => {
         );
         expect(document.querySelectorAll(".Mui-selected")).toHaveLength(selected.length);
     });
+    describe("Edit Mode", () => {
+        it("displays the data with edit buttons", async () => {
+            const dispatch = jest.fn();
+            const state: TaipyState = INITIAL_STATE;
+            const { getAllByTestId, queryAllByTestId, rerender } = render(
+                <TaipyContext.Provider value={{ state, dispatch }}>
+                    <PaginatedTable data={undefined} columns={editableColumns} onEdit="onEdit" showAll={true} />
+                </TaipyContext.Provider>
+            );
+
+            rerender(
+                <TaipyContext.Provider value={{ state: { ...state }, dispatch }}>
```

Comment:
Is a copy needed here?

Reply:
yes, otherwise the state doesn't change and there's no re-render

Reply:
But of course! Thx